src/modules/kvs/kvs.c

```diff
@@ -1342,6 +1342,46 @@ static int error_event_send (kvs_ctx_t *ctx, json_t *names, int errnum)
     return rc;
 }
 
+/* Optimization: the current rootdir object is optionally included
+ * in the kvs.setroot event.  Prime the local cache with it.
+ * If there are complications, just skip it.  Not critical.
+ */
+static void prime_cache_with_rootdir (kvs_ctx_t *ctx, json_t *rootdir)
+{
+    struct cache_entry *hp;
+    href_t ref;
+    void *data = NULL;
+    int len;
+
+    if (treeobj_validate (rootdir) < 0 || treeobj_is_dir (rootdir)) {
```

Comment:
```!treeobj_is_dir()```?

Reply:
Just ensuring that the "rootdir" really is a dir, not some other object.  Unlikely since the master would have already checked that.  Drop?

Reply:
I was thinking it should it be opposite logic.  If is not-dir, then error out.  Right now it errors if it is a dir.

Reply:
Oh, reverse logic, got it!